Firestore/core/src/firebase/firestore/auth/token.h

```diff
@@ -55,21 +55,22 @@ class Token {
    * state on disk, etc.).
    */
   const User& user() const {
+    FIREBASE_ASSERT(is_valid_);
```

Comment:
Code somewhere actually obtains user from invalid token; seems that's why you also changed empty-credential-provider to passing-in a valid token with invalid content. I guess we first need to come with what exactly `Token` will be, in real world and in test cases. That can help define the logic better. Say, can we make the following assumption:

In real world:
* Tried to get token for anonymous user: token obj is invalid and both token-string and user are undefined (i.e. the caller should check the validity and not depends on the token-string and user);
* Get token failed for a sign-in user: token obj is invalid and token-string is undefined and user are sign-in (possible? do we ever need to recover user from invalid token.);
* Get token succeeded for a sign-in user: token obj is valid and token string is non-empty and user is sign-in;
* Could Auth ever return a valid token for anonymous user?

In test cases: all of above besides
* Tried to get token for anonymous user: token-string is undefined or empty string whatever to make the header fallback to default in test and user is anonymous. Should this obj be valid or invalid?